core/problems/problems.go

```diff
@@ -0,0 +1,94 @@
+package problems
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/otrego/clamshell/core/board"
+	"github.com/otrego/clamshell/core/color"
+	"github.com/otrego/clamshell/core/movetree"
+	"github.com/otrego/clamshell/core/sgf"
+)
+
+// Flatten takes a Treepath and a MoveTree and returns
+// the root of the flat movetree. the flat movetree
+// ignores the last two nodes of the treepath
+func Flatten(tp movetree.Treepath, g *movetree.MoveTree) (*movetree.MoveTree, error) {
+	b, err := PopulateBoard(tp, g)
+	placements := b.GetFullBoardState()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var sbSGF strings.Builder
+	var sbAB strings.Builder
+	var sbAW strings.Builder
```

Comment:
Ah, you're constructing an SGF manually. I would make edits on the actual node rather than string-construction it should be more robust.

Reply:
Gotcha, should I use a new node, or can I manipulate the original movetree?

also, should the flat movetree point to the children of the node that we flatten to? I'm not sure if that's the way to communicate this.

do we want 
`flatTree.Root.Children = copy(originalTree.NodeFromTreepath(tp).Children)`
or do we want flatTree to have no children after Root